src/vm/jithelpers.cpp

```diff
@@ -5127,6 +5127,24 @@ HCIMPL0(void, JIT_ThrowPlatformNotSupportedException)
 }
 HCIMPLEND
 
+/*********************************************************************/
+HCIMPL0(void, JIT_ThrowArgTypeNotSupportedException)
+{
+    FCALL_CONTRACT;
+
+    /* Make no assumptions about the current machine state */
+    ResetCurrentContext();
+
+    FC_GC_POLL_NOT_NEEDED();    // throws always open up for GC
+
+    HELPER_METHOD_FRAME_BEGIN_ATTRIB_NOPOLL(Frame::FRAME_ATTR_EXCEPTION);    // Set up a frame
+
+    COMPlusThrow(kNotSupportedException, W("Arg_TypeNotSupported"));
```

Comment:
This behavior is slightly different from `S.N.Vector<T>` that throws NSE with message "Specified type is not supported".

https://github.com/dotnet/coreclr/pull/15028